qiskit_optimization/runtime/vqe_program.py

```diff
@@ -0,0 +1,378 @@
+        vqe_result.eigenvalue = result.get("eigenvalue", None)
+        vqe_result.aux_operator_eigenvalues = result.get("aux_operator_eigenvalues", None)
+        vqe_result.optimal_parameters = result.get("optimal_parameters", None)
+        vqe_result.optimal_point = result.get("optimal_point", None)
+        vqe_result.optimal_value = result.get("optimal_value", None)
+        vqe_result.optimizer_evals = result.get("optimizer_evals", None)
+        vqe_result.optimizer_time = result.get("optimizer_time", None)
+        vqe_result.optimizer_history = result.get("optimizer_history", None)
+
+        return vqe_result
+
+
+class VQEProgramResult(VQEResult):
+    """The VQEProgram result object.
+
+    This result objects contains the same as the VQEResult and additionally the history
+    of the optimizer, containing information such as the function and parameter values per step.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._job_id: Optional[str] = None
+        self._optimizer_history: Optional[Dict[str, Any]] = None
+
+    @property
+    def job_id(self) -> str:
+        """The job ID associated with the VQE runtime job."""
+        return self._job_id
+
+    @job_id.setter
+    def job_id(self, job_id: str) -> None:
+        """Set the job ID associated with the VQE runtime job."""
+        self._job_id = job_id
+
+    @property
+    def optimizer_history(self) -> Optional[Dict[str, Any]]:
+        """The optimizer history."""
+        return self._optimizer_history
+
+    @optimizer_history.setter
+    def optimizer_history(self, history: Dict[str, Any]) -> None:
+        """Set the optimizer history."""
+        self._optimizer_history = history
+
+
+def _validate_optimizer_settings(settings):
+    name = settings.get("name", None)
+    if name not in ["SPSA", "QN-SPSA"]:
+        raise NotImplementedError("Only SPSA and QN-SPSA are currently supported.")
+
+    allowed_settings = [
+        "name",
+        "maxiter",
+        "blocking",
+        "allowed_increase",
+        "trust_region",
+        "learning_rate",
+        "perturbation",
+        "resamplings",
+        "last_avg",
+        "second_order",
+        "hessian_delay",
+        "regularization",
+        "initial_hessian",
+    ]
+
+    if name == "QN-SPSA":
+        allowed_settings.remove("trust_region")
+        allowed_settings.remove("second_order")
+
+    unsupported_args = set(settings.keys()) - set(allowed_settings)
+
+    if len(unsupported_args) > 0:
+        raise ValueError(
+            f"The following settings are unsupported for the {name} optimizer: "
+            f"{unsupported_args}"
+        )
+
+
+def _convert_to_paulisumop(operator):
+    """Attempt to convert the operator to a PauliSumOp."""
+    if isinstance(operator, PauliSumOp):
+        return operator
+
+    try:
+        primitive = SparsePauliOp(operator.primitive)
+        return PauliSumOp(primitive, operator.coeff)
+    except Exception as exc:
```

Comment:
Can we catch any more specific exception here?

Reply:
I'm not sure what exceptions should come up here, so I just catched any and gave more context. The alternative would be to not try-except at all and just let the code fail without an extra message

Reply:
I see. We may keep it. But when it is obvious which kind of exceptions we expect, it is best to specify them and handle accordingly. In the above, we may have some wild exception and it would actually inform the user that the type of the operator is invalid, which may not be the problem...

Reply:
Yeah that's why I'm re-raising the original one, so that the information is not lost 🙂